Source/IGListBindingSectionController.m

```diff
@@ -29,8 +30,29 @@ @interface IGListBindingSectionController()
 
 @end
 
+static NSArray *objectsWithDuplicateIdentifiersRemoved(NSArray<id<IGListDiffable>> *objects) {
+    if (objects == nil) {
+        return nil;
+    }
+    
+    NSMutableSet *identifiers = [NSMutableSet new];
+    NSMutableArray *uniqueObjects = [NSMutableArray new];
+    for (id<IGListDiffable> object in objects) {
+        id diffIdentifier = [object diffIdentifier];
+        if (diffIdentifier != nil
+            && ![identifiers containsObject:diffIdentifier]) {
+            [identifiers addObject:diffIdentifier];
+            [uniqueObjects addObject:object];
+        } else {
+            IGLKLog(@"WARNING: Object %@ already appeared in objects array", object);
+        }
+    }
+    return uniqueObjects;
+}
```

Comment:
Since this is the same as the method already in `IGListAdapterUpdater`, how about moving this function to its own file in the `Internal` dir and sharing it between the updater and this? Then:

- If code ever changes, both places get the change/fix
- We can unit test this method directly

Reply:
I have tried that earlier but I was getting bunch of odd C99 (redeclaration) errors. I will give it a second shot over the weekend (probably Sunday)

Reply:
@rnystrom Do you have any naming conventions for such files? Currently, I am going for `IGListArrayUtilsInternal.h`

Reply:
Let me know, if you are happy with today's changes